### ISSUE TITLE
frame-src can leak cross origin information

### ISSUE THREAD

**Issue by antosart** [w3c/webappsec-csp#509]
CSP frame-src is set by the parent document, yet it controls navigations which might be initiated by other contexts (like the child iframe navigating itself). CSP violations caused by it get reported to the parent document, potentially leaking cross-origin information.

I think we should restrict frame-src to only control the iframe's src attribute, which is set and controlled by the parent.

**Comment by annevk** [comment]
At least per the specification this concern also applies to `child-src` and `object-src`, right? I agree that this is a problem.

**Comment by antosart** [comment]
You're right, I think this also applies to `object-src`. Regarding `child-src`, it applies only when it's a fallback for `frame-src`. I don't think there is an analogous problem for workers.

**Comment by antosart** [comment]
And if we want to cover all possible outgoing requests, I guess other types of navigations of iframes (e.g. by setting `window.location`) would be better governed by `navigate-to` (which would check the CSP of the initiator of the navigation and report an eventual violation there)

**Comment by annevk** [comment]
To be clear, navigate-to (and form-action) are scoped to the document that declares the policy. I don't think they should apply to other documents. (So in your example the frame would have to declare its own policy.)

**Comment by antosart** [comment]
As far as I understand, as per the specification they do (and I believe in chrome's implementation too). That's another issue though :)
